src/lib.rs

```diff
@@ -1244,45 +1249,59 @@ impl GooseAttack {
 
     // Configure how many Goose Users to hatch.
     fn set_users(&mut self) -> Result<(), GooseError> {
+        // Track how value gets set so we can return a meaningful error if necessary.
+        let mut key = "configuration.users";
+        let mut value = 0;
+
         // Check if --users is set.
         if let Some(users) = self.configuration.users {
-            // Setting --users with --worker is not allowed.
-            if self.attack_mode == GooseMode::Worker {
-                return Err(GooseError::InvalidOption {
-                    option: "--users".to_string(),
-                    value: users.to_string(),
-                    detail: "The --users option can not be set together with the --worker flag."
-                        .to_string(),
-                });
-            }
-
-            // Setting --users to 0 is not allowed.
-            if users == 0 {
-                return Err(GooseError::InvalidOption {
-                    option: "--users".to_string(),
-                    value: "0".to_string(),
-                    detail: "The --users option must be set to at least 1.".to_string(),
-                });
-            }
+            key = "--users";
+            value = users;
         // If not, check if a default for users is set.
         } else if let Some(default_users) = self.defaults.users {
             // On Worker hatch_rate comes from the Manager.
             if self.attack_mode == GooseMode::Worker {
                 self.configuration.users = None;
             // Otherwise use default.
             } else {
+                key = "set_default(GooseDefault::Users)";
+                value = default_users;
+
                 self.configuration.users = Some(default_users);
             }
         // If not and if not running on Worker, default to 1.
         } else if self.attack_mode != GooseMode::Worker {
+            value = self.number_of_cpus;
```

Comment:
I think this should not be needed.

Reply:
I don't understand?

If `--users` is set, we use that.
Otherwise, if `set_default(Users)` is set, we use that.
Otherwise we default to the number of cpus. 

Are you saying we shouldn't default to the number of cpus? Or that the `self.number_of_cpus` variable isn't needed? (The latter is correct: this was originally added because I was unable to call the function "on the fly" from a StructOpt structure -- now that we're not doing that, it can indeed be removed. Feels like it's better handled in another PR though?)

Reply:
Oh, the `value` shouldn't be needed. I agree it shouldn't be, but: 1) in Rust we have to set `value` to something in all code-paths, and 2) what if the library that tells us the number of cpus returns an invalid number (ie, 0)?

Reply:
I've added the missing `key`, and documented why we set up debug here even though failure isn't currently possible.